src/mapping/Mapping.cpp

```diff
@@ -78,6 +79,106 @@ int Mapping::getDimensions() const
   return _dimensions;
 }
 
+void Mapping::scaleConsistentMapping(int inputDataID, int outputDataID) const
+{
+  // If rank is not empty and do not contain connectivity information, raise error
+  if (input()->edges().empty() and (not input()->vertices().empty())) {
+    logging::Logger _log{"mapping::Mapping"};
+    PRECICE_ERROR("There is no connectivity information defined for mesh " << input()->getName() << ". Scaled consistent mapping requires connectivity information.");
+  }
+  if (output()->edges().empty() and (not output()->vertices().empty())) {
+    logging::Logger _log{"mapping::Mapping"};
+    PRECICE_ERROR("There is no connectivity information defined for mesh " << output()->getName() << ". Scaled consistent mapping requires connectivity information.");
+  }
+
+  const auto &inputValues  = input()->data(inputDataID)->values();
+  auto &      outputValues = output()->data(outputDataID)->values();
+
+  int valueDimensions = input()->data(inputDataID)->getDimensions();
+  int meshDimensions  = input()->getDimensions();
+
+  // Integral is calculated on each direction separately
+  std::vector<double> integralInput(valueDimensions);
+  std::vector<double> integralOutput(valueDimensions);
```

Comment:
There is a lot of code duplication here.
We could refactor the integration into a separate helper function and move it into `src/mesh/utils.hpp`. This also allows to test this functionality in isolation and reuse it in the future.

Ideally, these lines would look as follows:

```suggestion
  std::vector<double> integralInput = mesh::integrate(inputMesh, inputValues);
  std::vector<double> integralOutput = mesh::integrateOwners(outputMesh, outputValues);
```